site-src/guides/tls.md

```diff
@@ -17,6 +17,14 @@ For Gateways, there are two connections involved:
 With Gateway API, TLS configuration of downstream and
 upstream connections is managed independently.
 
+Depending on the Listener Protocol, different TLS modes and Route types are supported.
+
+Listener Protocol | TLS Mode | Route Type Supported
+--- | --- | ---
+TLS | Passthrough | TLSRoute
```

Comment:
Should it also be possible to have protocol `TLS`, tls mode `Terminate`, with route type `TLSRoute`?

Reply:
Given the Slack discussion, I understood it as "No". As TLSRoute will need TLS routing information which would be terminated? But maybe im wrong.

Reply:
Nah sounds like I don't understand either. I think the bit I'm not grasping is what @howardjohn described in the thread in regards to the 3 protocols, `incoming protocol, routing protocol, and upstream protocol` and how those apply.

Reply:
Good questions. Maybe @howardjohn @robscott @youngnick can help?

Reply:
I think there are three protocols involved here:

Protocol to the Gateway - defined by the Listener
Protocol to be used for Routing - defined by the Route (selected by the Listener)
Protocol to be used upstream from the Gateway - defined by the upstream Service, usually the same as routing protocol

That's at least my interpretation of how the different pieces fit together. We likely need a follow up to better document these concepts.

Reply:
> I wonder then also how the AppProtocol field comes into play, meaning if you had a TCPProxy with TLS on the gateway and an AppProtocol of TLS would you terminate TLS at the gateway, then forward upstream TLS?

I would expect two valid configurations for AppProtocol=TLS personally:
* TLS=Passthrough, forward the original TLS
* TLS=Terminate, with BackendPolicy configuring to use TLS. This would terminate then originate a new TLS connection.

Reply:
So should I change and list both modes?

Reply:
I could be wrong, but I think this discussion is somewhat different than the kind of routes that are supported with each mode. So although I think this is important to document, I don't think we need to delay getting this PR in.

Reply:
SGTM. Nitpick changes coming in soon